src/api/rate_limiter.py

```diff
@@ -78,31 +80,81 @@ async def handle_requests(self):
             if request.method == "GET":
                 # keep track of the last time a request was made
                 last_time_request = datetime.now()
+
                 # actually send the GET request
-                resp = requests.get(request.url, cookies=request.cookies)
-                if resp.status_code != 200:
-                    self.logger.warning(
-                        "Request : GET %s + %s failed with response code %s",
+                try:
+                    resp = requests.get(request.url, cookies=request.cookies, timeout=DEFAULT_MAX_TIMEOUT)
```

Comment:
MAX_TIMEOUT isn't really clear, wouldn't `REQUEST_TIMEOUT` be better?

As for the `_max_retry` it would be better to:
- in the constructor `self._request_timeout = DEFAULT_REQUEST_TIMEOUT`
- then use it `requests.get(..., timeout=self._request_timeout)`

seems much more flexible!